### DISCUSSION TITLE
Digital signature

### DISCUSSION THREAD

**Discussion by pavolholes** [valinet/ExplorerPatcher#2353] [Ideas]
Hi all and Valentin,
thanks a lot for a great work and amazing project. Is it possible to make the `ep_setup.exe` in the releases signed with digital signature so the Publisher is filled?

Thank you.

**Comment by pyrates999** [comment]
That costs money.  Read here why:

https://codesigningstore.com/free-code-signing-certificate

From the article:

_The code signing vetting process requires verification of certain things (documents), for example, government-issued ID proof, driving license, passport, and the financial and non-financial document containing your full name. And, this vetting process takes one to five days for completing and requires a large staff and operational costs as well. Due to these reasons, it’s not possible to offer a free code signing certificate –just like there are no free EV & OV SSL Certificates._

So if you want it to be signed, you could suggest a way to donate here instead.

**Comment by pavolholes** [comment]
Thanks for reply @pyrates999 . Thanks to your comment I found what I need to do some research.

I found the [SigStore](https://www.sigstore.dev/) but that's not valid in Windows OS. But then in the Issues of the SigStore I found [SignPath Foundation](https://signpath.org/about/) mentioned and that might be the solution.
From the https://github.com/sigstore/fulcio/issues/250#issuecomment-1697130810:
> Disclaimer: I work for SignPath
> Just wanted to clarify that for approved OSS projects, the [SignPath Foundation](https://signpath.org/) provides free code signing certificates and of course the usage of [our services](https://about.signpath.io/) are also free.

From the [SignPath for Open Source projects](https://about.signpath.io/product/open-source) I have a feeling it would be doable for the Explorer Patcher since it's a really big project with many stars and widely used. But I believe the repository owner @valinet should contact them via email to get the approval.

**Comment by pyrates999** [comment]
We'll see what happens then.

**Comment by SirWobbyTheFirst** [comment]
Bumping this with a +1 for SignPath.org because now that 22631 (Build 4169) has been pushed out to general release, I am not going to whitelist the executable in an antivirus just because it cannot be signed.

**Comment by Dymaxion00** [comment]
Hi, another bump to this — this causes problems on any machine with Smartscreen enabled, and many more problems if the machine has had any security hardening applied.  It looks like this could be done at no cost, with just a bit of additional hassle in the release process.  It would be very highly appreciated!
